instrumentation/opentelemetry-instrumentation-sqlalchemy/src/opentelemetry/instrumentation/sqlalchemy/engine.py

```diff
@@ -251,13 +263,22 @@ def _before_cur_exec(
                         if self.commenter_options.get(k, True)
                     }
 
-                    statement = _add_sql_comment(statement, **commenter_data)
+                    if self.enable_attribute_commenter:
+                        statement = _add_sql_comment(
+                            statement, **commenter_data
+                        )
 
                 span.set_attribute(SpanAttributes.DB_STATEMENT, statement)
                 span.set_attribute(SpanAttributes.DB_SYSTEM, self.vendor)
                 for key, value in attrs.items():
                     span.set_attribute(key, value)
 
+                if (
+                    self.enable_commenter
+                    and not self.enable_attribute_commenter
+                ):
+                    statement = _add_sql_comment(statement, **commenter_data)
+
```

Comment:
Line 267 and this line look the same both have self.enable_commenter=True, but on line 267, self.enable_attribute_commenter=True, while in this condition, self.enable_attribute_commenter=False. So, what's the difference? Can we call _add_sql_comment once to make it clearer

Reply:
I'll see how I can rearrange and add comments to make it more clear!

Currently L247 checks `self.enable_commenter=True` and L267 checks `self.enable_attribute_commenter=True`. If so, we `_add_sql_comment` first so that both `db.statement` attribute and executed query statement include the sqlcomment.

This next part is unclear to me too now 😅 If at L247 `self.enable_commenter=False`, then `db.statement` gets set first without sqlcomment. Then at L276 we check again?! Not sure how the tests are passing.